presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ActualProperties.java

```diff
@@ -323,7 +322,16 @@ public ActualProperties build()
         {
             List<LocalProperty<VariableReferenceExpression>> localProperties = this.localProperties;
             if (unordered) {
-                localProperties = filteredCopy(this.localProperties, property -> !property.isOrderSensitive());
+                ImmutableList.Builder<LocalProperty<VariableReferenceExpression>> newLocalProperties = ImmutableList.builder();
+                for (LocalProperty<VariableReferenceExpression> property : this.localProperties) {
```

Comment:
How do we know G(unique) is the first in this case? and are we able to guarantee that?

Reply:
No, we do not need to guarantee this, as which one is earlier depends on other code which produce this property (in this case it's the property generation code for AssignUniqueId) and not related to this part. This code only needs to make sure that properties which will be invalid after removal is still valid